SharpPcap/LibPcap/LibPcapSafeNativeMethods.cs

```diff
@@ -411,6 +411,19 @@ internal static int pcap_breakloop(IntPtr /* pcap_t_* */ p)
             return UseWindows ? Windows.pcap_breakloop(p) : Unix.pcap_breakloop(p);
         }
 
+        /// <summary>
+        /// See https://www.tcpdump.org/manpages/pcap_set_protocol_linux.3pcap.html for details
+        /// Note: This is implemented to document why we don't expose access to this method currently.
+        /// If you have a need for this method please open an issue documenting your need.
+        /// </summary>
+        /// <param name="p"></param>
+        /// <param name="protocol"></param>
+        /// <returns></returns>
+        internal static int pcap_set_protocol_linux(IntPtr /* pcap_t* */ p, int protocol)
+        {
+            throw new NotImplementedException("It should not be used in portable code; instead, a filter should be specified with pcap_setfilter(3PCAP).");
```

Comment:
Use `InvalidOperationException` to indicate that "this should not be done", instead of `NotImplementedException` that indicates "this is not supported/implemented yet, but we may support it in the future".
```suggestion
            throw new InvalidOperationException("It should not be used in portable code; instead, a filter should be specified with pcap_setfilter(3PCAP).");
```